File: src/easy/accept_into_movie.zig
const expectEqual = @import("std").testing.expectEqual;

pub fn acceptIntoMovie(age: f32, isSupervised: bool) bool {
    if (age < 15) return isSupervised;
    return true;
}
test "Given age = 14 and isSupervised = true should admit into Movie Theatre" {
    try expectEqual(acceptIntoMovie(14, true), true);
}
test "Given age = 15 and isSupervised = true should admit into Movie Theatre" {
    try expectEqual(acceptIntoMovie(15, true), true);
}
test "Given age = 16 and isSupervised = true should admit into Movie Theatre" {
    try expectEqual(acceptIntoMovie(16, true), true);
}
test "Given age = 14 and isSupervised = false should admit into Movie Theatre" {
    try expectEqual(acceptIntoMovie(14, false), false);
}
test "Given age = 15 and isSupervised = false should admit into Movie Theatre" {
    try expectEqual(acceptIntoMovie(15, false), true);
}
test "Given age = 16 and isSupervised = false should admit into Movie Theatre" {
    try expectEqual(acceptIntoMovie(16, false), true);
}
test "Given age = 14.99999 and isSupervised = true should admit into Movie Theatre" {
    try expectEqual(acceptIntoMovie(14.99999, true), true);
}
test "Given age = 14.99999 and isSupervised = false should admit into Movie Theatre" {
    try expectEqual(acceptIntoMovie(14.99999, false), false);
}

File: src/easy/accept_into_movie.rs
pub fn accept_into_movie(age: f32, is_supervised: bool) -> bool {
    match age < 15.0 {
        true => is_supervised,
        false => true,
    }
}
#[cfg(test)]
mod tests {
    use super::accept_into_movie;
    #[test]
    fn test_accept_into_movie() {
        assert_eq!(accept_into_movie(14.0, true), true);
        assert_eq!(accept_into_movie(15.0, true), true);
        assert_eq!(accept_into_movie(16.0, true), true);
        assert_eq!(accept_into_movie(14.0, false), false);
        assert_eq!(accept_into_movie(15.0, false), true);
        assert_eq!(accept_into_movie(16.0, false), true);
        assert_eq!(accept_into_movie(14.99999, true), true);
        assert_eq!(accept_into_movie(14.99999, false), false);
    }
}
